code-dot-org/code-dot-org
# Make studio.code.org/s/express-2021/lessons/1/levels/2 work offline

We want to setup studio.code.org/s/express-2021/lessons/1/levels/2 to able to work while the user is offline. This is currently hidden behind an experiment cookie in the user's browser.

## Links
* [JIRA](https://codedotorg.atlassian.net/browse/FND-1918) Initial implementation.
* [JIRA](https://codedotorg.atlassian.net/browse/FND-1908) Add retries and cleanup old data.

## Testing story
* Manually tested
  * Enable chrome://flags/#allow-insecure-localhost if you are testing on localhost.
  * Visit http://localhost-studio.code.org:3000/offline/join_pilot
  * Visit http://localhost-studio.code.org:3000/s/express-2021/lessons/1/levels/2
  * Open the developer tools and go into offline mode.
  * Refresh the page and observe that you can still complete the level.
